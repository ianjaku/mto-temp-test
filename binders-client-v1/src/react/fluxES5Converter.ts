/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
