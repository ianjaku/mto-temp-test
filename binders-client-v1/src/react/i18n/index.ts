import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
