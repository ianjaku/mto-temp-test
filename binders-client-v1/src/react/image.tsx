/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
