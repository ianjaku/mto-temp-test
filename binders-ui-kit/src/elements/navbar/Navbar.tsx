import * as React from "react";
import { Theme, createStyles, makeStyles } from "@material-ui/core/styles";
import Thumbnail, { FitBehaviour } from "../thumbnail";
import { useRibbonsBottomHeight, useRibbonsTopHeight } from "../../compounds/ribbons/hooks";
import ClientThumbnail from "@binders/client/lib/clients/repositoryservice/v3/Thumbnail";
import Hidden from "@material-ui/core/Hidden";
import MaterialDrawer from "@material-ui/core/Drawer";
import NavbarMenuItemElement from "./NavbarMenuItemElement";
import classnames from "classnames";
import "./navbar.styl";

export interface INavbarDrawerProps {
    className: string;
    elevation?: number;
    isOpen: boolean;
    onClose: () => void;
}

export const navbarDrawerWidth = 240;

const useStyles = makeStyles((theme: Theme) => createStyles({
    drawer: {
        [theme.breakpoints.up("sm")]: {
            flexShrink: 0,
            width: navbarDrawerWidth,
        },
    },
    drawerPaper: {
        width: navbarDrawerWidth,
    },
}));

export const NavbarDrawer = React.forwardRef<HTMLElement, React.PropsWithChildren<INavbarDrawerProps>>((props, ref) => {
    const classes = useStyles();
    const { className, elevation: propsElevation, isOpen, onClose } = props;
    const elevation = propsElevation || 8;

    return (
        <nav className={classes.drawer} ref={ref}>
            <Hidden smUp={true} implementation="css">
                <MaterialDrawer
                    anchor="left"
                    classes={{ root: className, paper: classes.drawerPaper }}
                    ModalProps={{ keepMounted: true }}
                    onClose={onClose}
                    open={isOpen}
                    variant="persistent"
                    PaperProps={{ elevation }}
                >
                    {props.children}
                </MaterialDrawer>
            </Hidden>
            <Hidden xsDown={true} implementation="css">
                <MaterialDrawer
                    open={true}
                    variant="permanent"
                    classes={{ root: className, paper: classes.drawerPaper }}
                    PaperProps={{ elevation }}
                >
                    {props.children}
                </MaterialDrawer>
            </Hidden>
        </nav>
    );
});

export interface INavbarMenuItem {
    label: string;
    link?: string;
    exact?: boolean;
    type: NavbarMenuItemType;
    element?: React.ReactElement;
}

export enum NavbarMenuItemType {
    advanced,
    analytics,
    create,
    dashboard,
    home,
    help,
    myLibrary,
    networkCheck,
    performance,
    pieChart,
    reader,
    settings,
    trash,
    users,
}

const Navbar: React.FC<{
    activateMenuItem?: (type: NavbarMenuItemType) => void;
    bottomItems: INavbarMenuItem[];
    headerElement?: React.ReactElement;
    headerImage?: ClientThumbnail;
    helpAccountComponent?: React.ReactElement;
    isMobileDrawerOpen?: boolean;
    items: INavbarMenuItem[];
    preselectedType?: NavbarMenuItemType;
    setIsMobileDrawerOpen?: (value: boolean) => void;
}> = ({
    activateMenuItem,
    bottomItems,
    children,
    headerElement,
    headerImage,
    helpAccountComponent,
    isMobileDrawerOpen,
    items,
    preselectedType,
    setIsMobileDrawerOpen,
}) => {
    const [activeMenuItemType, setActiveMenuItemType] = React.useState(preselectedType);
    const ribbonsTopHeight = useRibbonsTopHeight();
    const ribbonsBottomHeight = useRibbonsBottomHeight();

    const onMenuItemActivation = React.useCallback((type: NavbarMenuItemType) => {
        activateMenuItem?.(type);
        setActiveMenuItemType(type);
    }, [activateMenuItem]);

    const menuItemsWithoutCreate = React.useMemo(() => items.filter(i => i.type !== NavbarMenuItemType.create), [items]);
    const menuItemCreate = React.useMemo(() => items.find(i => i.type === NavbarMenuItemType.create), [items]);

    return (
        <div
            className={classnames("navbar", "navbar-left")}
            style={{
                paddingTop: `${ribbonsTopHeight}px`,
                paddingBottom: `${ribbonsBottomHeight}px`,
            }}
        >
            <div className="navbar-logo">
                <a href="/">
                    {headerImage && (
                        <Thumbnail
                            src={headerImage.buildRenderUrl({ requestedFormatNames: ["medium"] })}
                            fitBehaviour={headerImage.fitBehaviour === "crop" ? FitBehaviour.CROP : FitBehaviour.FIT}
                            bgColor={headerImage.bgColor}
                            imgClassName="navbar-logo-image"
                            width={80}
                            rotation={parseInt(headerImage.rotation, 10)}
                        />
                    )}
                </a>
            </div>
            <div className="navbar-main">
                <div className="navbar-main-navigation">
                    <ul className="navbar-main-navigation-list">
                        {headerElement}
                        {menuItemCreate?.element ?? null}
                        {menuItemsWithoutCreate.map(item => (
                            <NavbarMenuItemElement
                                key={item.type.toString()}
                                item={item}
                                active={item.type === activeMenuItemType}
                                onActivate={onMenuItemActivation}
                            />
                        ))}
                    </ul>
                </div>
                <div className="navbar-main-bottomsection">
                    <div className="navbar-main-navigation-list">
                        {bottomItems.map(item => (
                            <NavbarMenuItemElement
                                key={item.type.toString()}
                                item={item}
                                active={item.type === activeMenuItemType}
                                onActivate={onMenuItemActivation}
                            />
                        ))}
                        {helpAccountComponent}
                        {children}
                    </div>
                </div>
            </div>
            {isMobileDrawerOpen && <div className="navbar-outside-click" onClick={() => setIsMobileDrawerOpen?.(false)} />}
        </div>
    );
}

export default Navbar;
