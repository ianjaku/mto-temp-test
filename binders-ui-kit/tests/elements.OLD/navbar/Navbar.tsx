import * as React from "react";
import Navbar, { INavbarMenuItem, NavbarMenuItemType } from "../../../src/elements/navbar";
import { BrowserRouter } from "react-router-dom";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const NAVBAR_MENUITEM1 = { label: "menu item 1", link: "/", type: NavbarMenuItemType.dashboard } as INavbarMenuItem;
const NAVBAR_MENUITEM2 = { label: "menu item 2", link: "/users", type: NavbarMenuItemType.create } as INavbarMenuItem;
const NAVBAR_MENUITEMS = [ NAVBAR_MENUITEM1, NAVBAR_MENUITEM2 ];

const getMenuItems = () => NAVBAR_MENUITEMS;
let activeMenuItem;

const activateMenuItem = (type) => { activeMenuItem = type; };

const THUMBNAIL = {
    bgColor: "#ffffff",
    buildRenderUrl: () => Function.prototype(),
    fitBehaviour: "crop",
    medium: "https://dummyimage.com/400/400/fff",
};

const createDefaultNavbar = () => wrapWithTheme(
    <section>
        <BrowserRouter>
            <Navbar
                items={getMenuItems()}
                headerImage={THUMBNAIL}
                preselectedType={NavbarMenuItemType.create}
                activateMenuItem={activateMenuItem}
            />
        </BrowserRouter>
    </section>
);

// snapshots
test("Navbar (snapshot)", () => {
    const navbar =  create(createDefaultNavbar());
    const serialized = navbar.toJSON();
    expect(serialized).toMatchSnapshot();
});

// enzyme
test("Navbar (mount: enzyme+jsdom)", () => {
    const ddMount = mount(createDefaultNavbar());
    const navList = ddMount.find(".navbar-main-navigation-list").first();
    expect(navList.find("li")).toHaveLength(NAVBAR_MENUITEMS.length);
    const firstItem = navList.find("li").at(0);
    const secondItem = navList.find("li").at(1);
    expect(secondItem.find(".navbar-main-navigation-list-item-label").first().text()).toEqual(NAVBAR_MENUITEM2.label);

    expect(firstItem.hasClass("active")).toEqual(false);
    expect(secondItem.hasClass("active")).toEqual(true);
    expect(activeMenuItem).toEqual(undefined);

    firstItem.simulate("click");

    expect(ddMount.find(".navbar-main-navigation-list li").at(0).hasClass("active")).toEqual(true);
    expect(ddMount.find(".navbar-main-navigation-list li").at(1).hasClass("active")).toEqual(false);
    expect(activeMenuItem).toEqual(NAVBAR_MENUITEM1.type);
});
