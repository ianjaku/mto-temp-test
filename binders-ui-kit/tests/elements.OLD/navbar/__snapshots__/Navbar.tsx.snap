// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar (snapshot) 1`] = `
<section>
  <div
    className="navbar navbar-left"
  >
    <div
      className="navbar-logo"
    >
      <a
        href="/"
      >
        <div
          className="thumbnail-outer-wrapper"
          style={
            Object {
              "backgroundColor": "##ffffff",
              "height": undefined,
              "width": undefined,
            }
          }
        >
          <div
            className="thumbnail-wrapper crop landscape"
            onClick={undefined}
            onDoubleClick={undefined}
            onMouseDownCapture={undefined}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "##ffffff",
                "height": undefined,
                "width": undefined,
              }
            }
          >
            <img
              className="thumbnail crop landscape navbar-logo-image"
              onLoad={[Function]}
              src="https://dummyimage.com/400/400/fff"
              style={
                Object {
                  "height": undefined,
                  "left": "50%",
                  "position": "absolute",
                  "top": "50%",
                  "transform": "translate(-50%, -50%)",
                  "width": "auto",
                }
              }
              title="https://dummyimage.com/400/400/fff"
            />
          </div>
        </div>
      </a>
    </div>
    <div
      className="navbar-main"
    >
      <div
        className="navbar-main-navigation"
      >
        <ul
          className="navbar-main-navigation-list"
        >
          <li
            className="navbar-main-navigation-list-item"
            onClick={[Function]}
          >
            <a
              aria-current={null}
              className="navbar-main-navigation-list-item-link"
              href="/"
              onClick={[Function]}
              style={undefined}
            >
              <svg
                className="navbar-main-navigation-list-item-icon"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                />
              </svg>
              <label
                className="navbar-main-navigation-list-item-label"
              >
                menu item 1
              </label>
            </a>
          </li>
          <li
            className="navbar-main-navigation-list-item active"
            onClick={[Function]}
          >
            <a
              aria-current={null}
              className="navbar-main-navigation-list-item-link"
              href="/users"
              onClick={[Function]}
              style={undefined}
            >
              <svg
                className="navbar-main-navigation-list-item-icon"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "currentColor",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9h-4v4h-2v-4H9V9h4V5h2v4h4v2z"
                />
              </svg>
              <label
                className="navbar-main-navigation-list-item-label"
              >
                menu item 2
              </label>
            </a>
          </li>
        </ul>
      </div>
      <div
        className="navbar-main-bottomsection"
      />
    </div>
  </div>
</section>
`;
